File: app/src/main/java/com/jgm90/cloudmusic/dialogs/PlaylistDialog.java
package com.jgm90.cloudmusic.dialogs;

import android.content.Context;
import android.view.View;
import android.widget.EditText;

import androidx.annotation.NonNull;

import com.afollestad.materialdialogs.DialogAction;
import com.afollestad.materialdialogs.MaterialDialog;
import com.jgm90.cloudmusic.R;
import com.jgm90.cloudmusic.data.PlaylistData;
import com.jgm90.cloudmusic.interfaces.DialogCaller;
import com.jgm90.cloudmusic.models.PlaylistModel;

public class PlaylistDialog {

    private Context context;
    private MaterialDialog.Builder dialog_builder;
    private MaterialDialog dialog;
    private View dv;
    private DialogCaller listener;
    private EditText txt_playlist_name;
    private PlaylistModel obj;
    private PlaylistData dao;
    private boolean cargado;

    public PlaylistDialog(Context context, DialogCaller listener) {
        this.context = context;
        this.listener = listener;
        this.dao = new PlaylistData(context);
    }

    public void show() {
        dialog_builder = new MaterialDialog.Builder(context)
                .title("Playlist")
                .customView(R.layout.dialog_new_playlist, false)
                .cancelable(false)
                .autoDismiss(false)
                .positiveText("OK")
                .negativeText("CANCEL")
                .onPositive(new MaterialDialog.SingleButtonCallback() {
                    @Override
                    public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
                        if (validate()) {
                            guardar();
                            dialog.dismiss();
                            listener.onPositiveCall();
                        }
                    }
                })
                .onNegative(new MaterialDialog.SingleButtonCallback() {
                    @Override
                    public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
                        dialog.dismiss();
                    }
                });
        dialog = dialog_builder.build();
        dialog.show();
        dv = dialog.getCustomView();
        bind();
    }

    public void bind() {
        txt_playlist_name = dv.findViewById(R.id.txt_playlist_name);
        if (cargado) {
            txt_playlist_name.setText(obj.getName());
        }
    }

    public boolean validate() {
        if (txt_playlist_name.getText().toString().length() > 0) {
            return true;
        }
        return false;
    }

    public void guardar() {
        PlaylistModel item = new PlaylistModel(
                0,
                txt_playlist_name.getText().toString(),
                0
        );
        if (cargado) {
            item.setPlaylist_id(obj.getPlaylist_id());
            dao.update(item);
        } else {
            dao.insert(item);
        }
    }

    public void cargar(PlaylistModel obj) {
        this.obj = obj;
        cargado = true;
        show();
    }
}

File: app/src/main/java/com/jgm90/cloudmusic/dialogs/PlaylistDialog.kt
package com.jgm90.cloudmusic.dialogs

import android.content.Context
import android.view.View
import android.widget.EditText
import com.afollestad.materialdialogs.DialogAction
import com.afollestad.materialdialogs.MaterialDialog
import com.jgm90.cloudmusic.R
import com.jgm90.cloudmusic.data.PlaylistData
import com.jgm90.cloudmusic.interfaces.DialogCaller
import com.jgm90.cloudmusic.models.PlaylistModel

class PlaylistDialog(private val context: Context, private val listener: DialogCaller?) {
    private val dao: PlaylistData = PlaylistData(context)
    private var dialog: MaterialDialog? = null
    private var dialogView: View? = null
    private var txtPlaylistName: EditText? = null
    private var obj: PlaylistModel? = null
    private var cargado = false

    fun show() {
        val dialogBuilder = MaterialDialog.Builder(context)
            .title("Playlist")
            .customView(R.layout.dialog_new_playlist, false)
            .cancelable(false)
            .autoDismiss(false)
            .positiveText("OK")
            .negativeText("CANCEL")
            .onPositive(object : MaterialDialog.SingleButtonCallback {
                override fun onClick(dialog: MaterialDialog, which: DialogAction) {
                    if (validate()) {
                        guardar()
                        dialog.dismiss()
                        listener?.onPositiveCall()
                    }
                }
            })
            .onNegative(object : MaterialDialog.SingleButtonCallback {
                override fun onClick(dialog: MaterialDialog, which: DialogAction) {
                    dialog.dismiss()
                }
            })
        dialog = dialogBuilder.build()
        dialog?.show()
        dialogView = dialog?.customView
        bind()
    }

    private fun bind() {
        txtPlaylistName = dialogView?.findViewById(R.id.txt_playlist_name)
        if (cargado) {
            txtPlaylistName?.setText(obj?.name)
        }
    }

    private fun validate(): Boolean {
        return !txtPlaylistName?.text.isNullOrEmpty()
    }

    private fun guardar() {
        val item = PlaylistModel(
            0,
            txtPlaylistName?.text?.toString().orEmpty(),
            0,
        )
        if (cargado) {
            item.playlist_id = obj?.playlist_id ?: 0
            dao.update(item)
        } else {
            dao.insert(item)
        }
    }

    fun cargar(obj: PlaylistModel) {
        this.obj = obj
        cargado = true
        show()
    }
}
